arduino/smartcar/smartcar.ino

```diff
@@ -14,18 +18,77 @@ void setup() {
 }
 
 void loop() {
-  car.setSpeed(90);
-  preventCrash();
+  car.setSpeed(50); 
+  preventCrashWithUltrasonic(); 
+  preventCrashWithInfraredFront(); 
+  preventCrashWithInfraredLeft();
+  preventCrahsWithInfraredRight();
+  preventCrashWithInfraredBack();
 }
 
-void preventCrash() {
-  unsigned int fDistance = front.getDistance();
+void preventCrashWithUltrasonic() {
+  unsigned int fDistance = frontUS.getDistance();
   if (fDistance > 0 && fDistance < 200) {
+    Serial.print("Obstacle detected by front ultrasonic sensor");
     car.setSpeed(0);
     delay(500);
-    control.overrideMotorSpeed(100, -100);
+    control.overrideMotorSpeed(50, -100);
     delay(800);
     car.setSpeed(0);
     delay(200);
   }
 }
+
+void preventCrashWithInfraredFront() {
+  unsigned int fDistance = frontIR.getDistance();
+  if (fDistance > 0 && fDistance < 200) {
+    Serial.print("Obstacle detected by front infrared sensor"); 
+    car.setSpeed(0);
+    delay(500);
+    control.overrideMotorSpeed(50, -100);
+    delay(800);
+    car.setSpeed(0);
+    delay(200);
+  }
+}
+
+void preventCrashWithInfraredLeft() {
+  unsigned int fDistance = leftIR.getDistance();
+  if (fDistance > 0 && fDistance < 200) {
+    Serial.print("Obstacle detected by left infrared sensor"); 
+    car.setSpeed(0);
+    delay(500);
+    control.overrideMotorSpeed(50, -100);
```

Comment:
`control.overrideMotor` might work, but it's very unconventional.
Prefer to use the [overrideMotorSpeed](https://platisd.github.io/smartcar_shield/class_simple_car.html#ac25a30173ae4f42a785363997c401e95) function of the `Car` classes instead, that takes the same arguments and uses the higher abstraction (i.e. the `SimpleCar`).

Reply:
Done.